### ISSUE TITLE
.count() & paginate query performance issue

### ISSUE THREAD

**Issue by samhibberd** [ethercreative/simplemap#224]
### Description

We are running into up to 40 second query execution time when using a standard craft entry query with either the paginate tag or a count() query. The odd thing is that we currently only run into the issue when also querying using the `relatedTo` parameter. I have spent the day trying to eliminate bad coding our end, but can't pinpoint anything that might be causing this.

Site is large ish, 2k entries with 4 category groups fields which can make up the relations part of the query.

Querying `.all()` with limit set to 250 returns all results in 200ms and the same query of `.count()` took 43s, `.ids()` took 99ms.

```
{% set events = craft.entries()
	.section('events')
	.location({
		location: { lat: 51.060979388196934, lng: -1.3106352835893633 },
		radius: 100,
		unit: 'mi',
	})
	.relatedTo([
		'and',
		{ targetElement: [379], field: 'sports' },
		{ targetElement: [23890], field: 'distances' },
	])
	.limit(250)
	.count()
	%}
```

In all honesty i'm not sure if there is an issue here, or with the Craft core or with our install, either way we are out of ideas :( 

Happy to provide any additional detail / resources that would help recreate / debug further, I have also attached the sql of one of the slow queries if that helps.
 
[Slower.txt](https://github.com/ethercreative/simplemap/files/3894374/Slower.txt)

### Additional info

- Craft version: 3.3.16.1
- Maps version: 3.7.2
- PHP version: 7.2.20
- Database driver & version: MySQL 5.5.5

**Comment by Tam** [comment]
Count queries often, paradoxically, take longer than simply querying the data since they have to loop over every row in the database. With MySQL you _might_ be able to improve it by using `.count(1)`, but you're probably better off counting the `.ids()` (not that that's possible if you're using `paginate`).

Does the issue only occur when using both `location` and `relatedTo` in tandem?

If you're able to send a DB dump and your template code to dev@ethercreative.co.uk that would make it easier for me to debug.

**Comment by samhibberd** [comment]
Hi @Tam thanks for the feedback, I will email over my debugging progress. 

The scenario which consistently causes it for me, is when using `location` and `relatedTo`s across more than one category group.

**Comment by Tam** [comment]
Thanks for the db dump @samhibberd. Just released 3.7.4 which should vastly improve the query speed!

**Comment by samhibberd** [comment]
Great work, initial testing show 44s to 53ms ⚡️⚡️⚡️⚡️⚡️⚡️
